### ISSUE TITLE
Invalid request body when size is more than 30kb

### ISSUE THREAD

**Issue by chokami** [alicommit-malp/Dotnet-Core-Request-Body-Peeker#4]
Hi,
I had problems with large request bodies, it returns invalid json. I ended up changing the buffer with the following.

Original:
```C#
        public static async Task<string> PeekBodyAsync(this HttpRequest request, Encoding encoding = null)
        {
            try
            {
                if (encoding == null) encoding = new UTF8Encoding();
                request.EnableBuffering();
                var buffer = new byte[Convert.ToInt32(request.ContentLength)];
                if (buffer.Length == 0) return Empty;

                await request.Body.ReadAsync(buffer, 0, buffer.Length);
                return encoding.GetString(buffer);
            }
            finally
            {
                request.Body.Position = 0;
            }
        }
```

Updated in my codebase:
```C#
        public static async Task<string> ReadRequestBody(this HttpRequest request)
        {
            try
            {
                request.EnableBuffering();

                using (var reader = new StreamReader(
                    request.Body,
                    encoding: Encoding.UTF8,
                    detectEncodingFromByteOrderMarks: false,
                    leaveOpen: true))
                {
                    var bodyAsText = await reader.ReadToEndAsync();

                    return bodyAsText;
                }
            }
            finally
            {
                request.Body.Position = 0;
            }
        }}
```

**Comment by alicommit-malp** [comment]
Thanks , It is a good idea .
I will change the code accordingly and share the new version

**Comment by adamfisher** [comment]
I've added a PR to fix this. Any chance we can get this merged in soon? Would like to use the fixed version that properly reads the entire body.

**Comment by alicommit-malp** [comment]
Thanks for the PR let me have a look and get back to you :)

**Comment by alicommit-malp** [comment]
the issue has been addressed in the version 1.3.0 
https://www.nuget.org/packages/Request.Body.Peeker/1.3.0
